flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -388,16 +381,18 @@ public Task(
 			executionId,
 			this,
 			inputGateDeploymentDescriptors,
-			metrics.getIOMetricGroup(),
-			inputGroup,
-			buffersGroup);
+			taskShuffleMetricGroup);
 
 		this.inputGates = new InputGate[gates.length];
 		int counter = 0;
 		for (InputGate gate : gates) {
 			inputGates[counter++] = new InputGateWithMetrics(gate, metrics.getIOMetricGroup().getNumBytesInCounter());
 		}
 
+		// we will have to check type of shuffle service later whether it is NetworkEnvironment
+		//noinspection deprecation
+		networkEnvironment.registerLegacyNetworkMetrics(metrics.getIOMetricGroup(), resultPartitionWriters, gates);
```

Comment:
ditto: it seems a bit trouble after task only sees `ShuffleEnvironment` future, otherwise we need expose this method in interface.

Reply:
Since we intend to remove this in the next release It should be fine to use a cast here, so that we don't have to add such a short-lived method to the interface.

Reply:
which is now happening after rebasing onto #8608